analytic_engine/src/instance/flush_compaction.rs

```diff
@@ -167,6 +167,21 @@ pub struct TableFlushRequest {
     pub table_data: TableDataRef,
     /// Max id of memtable to flush (inclusive).
     pub max_memtable_id: MemTableId,
+    /// Max sequence number to flush (inclusive).
+    pub max_sequence: SequenceNumber,
+    /// Flush policy.
+    pub policy: TableFlushPolicy,
+}
+
+/// Policy of how to perform flush operation.
+#[derive(Debug)]
+pub enum TableFlushPolicy {
```

Comment:
Maybe this should be called `TableFlushAction` and the `Policy` may be reserved to for the trait to decide the `TableFlushRequest`, e.g.:
```rust
pub trait TableFlushPolicy {
    fn decide(&self, ...) -> Result<TableFlushRequest>;
}
```